va/disclosures.py

```diff
@@ -8,25 +8,73 @@
 
 class VirginiaDisclosureScraper(Scraper):
 
-    def scrape_committees(self):
-        SEARCH_COMMITTEES_URL="http://cfreports.sbe.virginia.gov/"
-        my_scraper = scrapelib.Scraper()
-        _, resp = my_scraper.urlretrieve(SEARCH_COMMITTEES_URL)
-        d = etree.fromstring(resp.content, parser=HTMLParser())
-        number_of_result_pages=int(d.xpath('//span[@id="PagingTotalPages"]/text()')[0])
-
-        target_table = d.xpath('//table/tbody')[0]
+    def parse_committee_table(self, target_table):
         scraped_rows = []
         for row in target_table.xpath('tr'):
-            data_dict = {}
             columns = row.xpath('td')
             name = columns[0].text_content().strip()
+            candidate = columns[1].text_content().strip()
+            committee_type = columns[2].text_content().strip()
+            anchors = columns[3].xpath('a')
+            try:
+                reports_link = anchors[0].attrib['href']
+            except IndexError:
+                reports_link = None
             if not (name == ""):
+                data_dict = {}
                 data_dict['org_name'] = name
+                data_dict['org_type'] = committee_type
+                if(committee_type == "Candidate Campaign Committee"):
+                    data_dict['org_candidate'] = candidate
+                data_dict['org_reports_link'] = reports_link
+                scraped_rows.append(data_dict)
+        return(scraped_rows)
+
+    def parse_scheduled_report_table(self, target_table):
+        scraped_rows = []
+        for row in target_table.xpath('tr'):
+            reporting_period = columns[0].text_content().strip()
+            amendment_number = columns[1].text_content().strip()
+            date_filed = columns[2].text_content().strip()
+            contributions_received = columns[3].text_content().strip()
+            ending_balance = columns[4].text_content().strip()
+            anchors = columns[5].xpath('a')
+            try:
+                report_link = anchors[0].attrib['href']
+            except IndexError:
+                report_link = None
+        if not (reporting_period == ""):
+                data_dict = {}
+                data_dict['rpt_period'] = reporting_period
+                data_dict['rpt_amendment'] = amendment_number
+                data_dict['rpt_filed'] = date_filed
+                data_dict['rpt_contribution'] = contributions_received
+                data_dict['rpt_balance'] = ending_balance
+                data_dict['rpt_link'] = report_link
                 scraped_rows.append(data_dict)
+        return(scraped_rows)
+
+    def scrape_committees(self):
+        SEARCH_COMMITTEES_URL="http://cfreports.sbe.virginia.gov/"
+
+        my_scraper = scrapelib.Scraper()
```

Comment:
so you don't need to initialize a new scraper, here.  Notice, on line 9 above, that the class you're working in is itself a scraper.  so, instead of `my_scraper.urlretrieve`, you can (and should) just call `self.urlretrieve`. The `VirginiaDisclosureScraper` that you're working on has all of the properties and methods that the scrapelib `Scraper` object has (plus a few others, including the ones you're adding).